dashboard/lib/services/lti.rb

```diff
@@ -92,28 +92,36 @@ def self.initialize_lti_student_from_nrps(client_id:, issuer:, nrps_member:)
     user
   end
 
-  def self.parse_nrps_response(nrps_response)
+  def self.parse_nrps_response(nrps_response, issuer)
     sections = {}
     members = nrps_response[:members]
     context_title = nrps_response.dig(:context, :title)
     members.each do |member|
       next if member[:status] == 'Inactive' || member[:roles].exclude?(Policies::Lti::CONTEXT_LEARNER_ROLE)
       # TODO: handle multiple messages. Shouldn't be needed until we support Deep Linking.
-      message = member[:message].first
 
-      # Custom variables substitutions must be configured in the LMS.
-      custom_variables = message[Policies::Lti::LTI_CUSTOM_CLAIMS.to_sym]
+      # If the LMS hasn't implemented the resource link level membership service, we don't get the message property in the member object
+      if Policies::Lti.issuer_accepts_resource_link?(issuer)
+        message = member[:message].first
 
-      # Handles the possibility of the LMS not having sectionId variable substitution configured.
-      member_section_ids = custom_variables[:section_ids]&.split(',') || [nil]
-      # :section_names from Canvas is a stringified JSON array
-      member_section_names = JSON.parse(custom_variables[:section_names])
+        # Custom variables substitutions must be configured in the LMS.
+        custom_variables = message[Policies::Lti::LTI_CUSTOM_CLAIMS.to_sym]
+
+        # Handles the possibility of the LMS not having sectionId variable substitution configured.
+        member_section_ids = custom_variables[:section_ids]&.split(',') || [nil]
+        # :section_names from Canvas is a stringified JSON array
+        member_section_names = JSON.parse(custom_variables[:section_names])
+
+      else
+        member_section_ids = [nrps_response.dig(:context, :id)]
+        member_section_names = [nil]
+      end
       member_section_ids.each_with_index do |section_id, index|
         if sections[section_id].present?
           sections[section_id][:members] << member
         else
           sections[section_id] = {
-            name: "#{context_title}: #{member_section_names[index]}",
+            name: member_section_names[index].nil? ? context_title.to_s : "#{context_title}: #{member_section_names[index]}",
```

Comment:
Nit: might be helpful to have a comment here explaining that Schoology already returns the context_title in the format want?